pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -198,11 +199,13 @@ func (r *Reconciler) handleUpdate(ctx context.Context, crp *fleetv1beta1.Cluster
 		return ctrl.Result{}, err
 	}
 
-	if err := r.Client.Status().Update(ctx, crp); err != nil {
-		klog.ErrorS(err, "Failed to update the status", "clusterResourcePlacement", crpKObj)
-		return ctrl.Result{}, err
+	if !apiequality.Semantic.DeepEqual(oldCRP.Status, crp.Status) {
```

Comment:
unfortuantely, this is not safe as we don't reconcile on status change

Reply:
I still don't quite understand:

- who else do you expect will update the status of this resource and make it out of sync?
- why can't a periodic resync that you already have course-correct drifts eventually?

Reply:
- Only the CRP controller changes its status but the reconcile is triggered by many other objects's status change. 
- if we rely on periodic resync to course-correct then it means the resync period needs to be reasonably short (say a min) which is something we agreed not a good practice.

We should still try to optimize the startup sequence though